### PR TITLE
feat: Argo CD lab with App of Apps pattern example

### LINKED ISSUE #403: [feat] Add advanced ArgoCD lab with App of Apps example using several applications from Sample
### What would you like to be added?

A an [ArgoCD](https://argo-cd.readthedocs.io/en/stable/) lab to the [Automation](https://www.eksworkshop.com/docs/automation) under the subsection [GitOps](https://www.eksworkshop.com/docs/automation/gitops/)

ArgoCD lab should show how Argo CD App of Apps pattern could be implemented to deliver complete set of components from Sample App to specific environment (dev) with specific customisations for that environment. 

### Why is this needed?

To show more advanced concept of GitOps and EKS

### PR DESCRIPTION
**Pull request by ybezsonov**
#### What this PR does / why we need it:
ArgoCD lab should show how Argo CD App of Apps pattern could be implemented to deliver complete set of components from Sample App to specific environment (dev) with specific customisations for that environment.

#### Which issue(s) this PR fixes:

Fixes #403

#### Quality checks

- [x] My content adheres to the style guidelines
- [x] I ran `make test` or `make e2e-test` and it was successful

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by csantanapr** [comment]
@candonov @tsahiduek @allamand I believe this PR is now ready to be review

**Comment by candonov** [comment]
@ybezsonov It look like this branch has conflicts, could you please resolve them?

File: website/docs/automation/gitops/argocd/app-of-apps/cleanup.md [hunk 1]
```diff
@@ -0,0 +1,10 @@
+---
```
Comment (niallthomson) [inline website/docs/automation/gitops/argocd/app-of-apps/cleanup.md:1]: Suggest moving this page in to the app of apps section itself as part of setting up

File: website/docs/automation/gitops/argocd/app-of-apps/index.md [hunk 2]
```diff
@@ -0,0 +1,113 @@
+---
```
Comment (niallthomson) [inline website/docs/automation/gitops/argocd/app-of-apps/index.md:1]: Can you limit this page to an introduction to the concept of app of apps and move the hands-on part in to a sub-page called "setup" or something similar?

File: website/docs/automation/gitops/argocd/app-of-apps/index.md [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+title: "Argo CD App of Apps"
```
Comment (niallthomson) [inline website/docs/automation/gitops/argocd/app-of-apps/index.md:2]: Remove "Argo CD"

File: website/docs/automation/gitops/argocd/app-of-apps/index.md [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+sidebar_position: 100
+---
+
+[Argo CD](https://argoproj.github.io/cd/) can deploy a set of applications to different environments (DEV, TEST, PROD ...) using common set of Kubernetes manifest and customizations specific to an environment.
+
+We can leverage [Argo CD App of Apps pattern](https://argo-cd.readthedocs.io/en/stable/operator-manual/cluster-bootstrapping/) to implement this use case. This pattern allows us to specify one Argo CD Application that consists of other applications.
+
+![argo-cd-app-of-apps](assets/argocd-app-of-apps.png)
+
+We reference [EKS Workshop Git repository](https://github.com/aws-samples/eks-workshop-v2/tree/main/environment/workspace/manifests) as a Git repository with Common Manifest. This repository will contain an initial resource state for each application.
```
Comment (niallthomson) [inline website/docs/automation/gitops/argocd/app-of-apps/index.md:13]: Unsure what "Common Manifest" means or why its capitalized

File: website/docs/automation/gitops/argocd/app-of-apps/index.md [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+`values.yaml` contains values which are specific for a particular environment and which will be applied to all application templates.
+
+```file
+automation/gitops/argocd/app-of-apps/values.yaml
+```
+
+First, copy `App of Apps` configuration which we described above to the Git repository directory:
+
+```bash
+$ cd ~/environment
```
Comment (niallthomson) [inline website/docs/automation/gitops/argocd/app-of-apps/index.md:74]: Lets remove the `cd` command and change the `yq` command to something like:

```
yq -i ".spec.source.repoURL = load(\"$HOME/environment/argocd_repo_url\")" ~/environment/argocd/app-of-apps/values.yaml
```

**Review by niallthomson — CHANGES_REQUESTED** [review]
These are some initial suggestions for structure, flow etc.

File: website/docs/automation/gitops/argocd/app-of-apps/index.md (hunk 2 above)
Comment (ybezsonov) [reply website/docs/automation/gitops/argocd/app-of-apps/index.md:1]: I created "Setup" section

File: website/docs/automation/gitops/argocd/app-of-apps/index.md (hunk 3 above)
Comment (ybezsonov) [reply website/docs/automation/gitops/argocd/app-of-apps/index.md]: done

File: website/docs/automation/gitops/argocd/app-of-apps/index.md (hunk 4 above)
Comment (ybezsonov) [reply website/docs/automation/gitops/argocd/app-of-apps/index.md]: Changed to the terms used in the workshop introduction: "base" manifests

File: website/docs/automation/gitops/argocd/app-of-apps/index.md (hunk 5 above)
Comment (ybezsonov) [reply website/docs/automation/gitops/argocd/app-of-apps/index.md]: changed

**Pull request merged**